src/sentry/utils/appleconnect/itunes_connect.py

```diff
@@ -3,349 +3,515 @@
+
+        This key is pretty static and can be cached long-term, saving a request to fetch it.
+        It can optionally be provided to :meth:`init`.
+
+        :returns: the service key to be used in all future calls X-Apple-Widget-Key headers.
+
+        :raises: any exception in case of failure.
+        """
+        url = "https://appstoreconnect.apple.com/olympus/v1/app/config?hostname=itunesconnect.apple.com"
+        logger.debug("GET %s", url)
+        response = self.session.get(url)
+        data = response.json()
+        return ITunesServiceKey(data["authServiceKey"])
+
+    def start_login_sequence(self, username: str, password: str) -> None:
+        """Starts the login process.
+
+        This will trigger the two factor authentication code to show on any validated device
+        for the user.  The code must be provided using :meth:`auth_code`.
+
+        Alternatively after calling this you can switch to SMS two factor authentication by
+        calling :meth:`request_sms`.
+        """
+        assert self.state in [ClientState.NEW, ClientState.EXPIRED]
+        url = "https://idmsa.apple.com/appleauth/auth/signin"
+        logger.debug("POST %s", url)
+
+        start_login = self.session.post(
+            url,
+            json={
+                "accountName": username,
+                "password": password,
+                "rememberMe": True,
+            },
+            headers={
+                "X-Requested-With": "XMLHttpRequest",
+                "X-Apple-Widget-Key": self.service_key,
+                "Accept": "application/json, text/javascript",
+            },
         )
 
-
-def get_dsym_url(
-    session: Session, app_id: str, bundle_short_version: str, bundle_version: str, platform: str
-) -> Optional[str]:
-    """
-    Returns the url for a dsyms bundle. The session must be logged in.
-    :return: The url to use for downloading the dsyms bundle
-    """
-    with sentry_sdk.start_span(
-        op="itunes-dsym-url", description="Request iTunes dSYM download URL"
-    ):
-        details_url = (
-            f"https://appstoreconnect.apple.com/WebObjects/iTunesConnect.woa/ra/apps/"
-            f"{app_id}/platforms/{platform}/trains/{bundle_short_version}/builds/"
-            f"{bundle_version}/details"
+        if start_login.status_code == http.HTTPStatus.OK:
+            raise ITunesError(
+                f"Two factor auth not enabled for user, status_code={start_login.status_code}"
+            )
+        elif start_login.status_code == http.HTTPStatus.CONFLICT:
+            self.session_id = start_login.headers["x-apple-id-session-id"]
+            self._scnt = start_login.headers["scnt"]
+            self.state = ClientState.AUTH_REQUESTED
+        elif start_login.status_code == http.HTTPStatus.UNAUTHORIZED:
+            raise InvalidUsernamePasswordError
+        else:
+            raise ITunesError(f"Unexpected status code form signing: {start_login.status_code}")
```

Comment:
i'm having a bit of trouble parsing this sentence: is this supposed to be `Unexpected status code from signing in`? or something else?